### PR TITLE
Fix #5417: VirusTotal proper error handling

### LINKED ISSUE #5417: FileUpload:  VirusTotal API limit error
By defaut VirusTotal have a rate limit (4 requests/minute) for non premium user.
I you try to scan more than 4 files in one minute you get a JSON exception because the response from Virus Total is not correctly managed 

```java
org.primefaces.json.JSONException: A JSONObject text must begin with '{' at 1 [character 2 line 1]
	at org.primefaces.json.JSONTokener.syntaxError(JSONTokener.java:433)
	at org.primefaces.json.JSONObject.<init>(JSONObject.java:196)
	at org.primefaces.json.JSONObject.<init>(JSONObject.java:320)
	at org.primefaces.virusscan.impl.VirusTotalVirusScanner.performVirusScan(VirusTotalVirusScanner.java:73)
	at org.primefaces.virusscan.VirusScannerService.performVirusScan(VirusScannerService.java:72)
	at org.primefaces.util.FileUploadUtils.performVirusScan(FileUploadUtils.java:280)
```

1) Environment
- PrimeFaces version: 7.0
- Application server + version: Wildlfy 17.0.1
- Affected browsers: Firefox / Chrome

## 2) Expected behavior

Have a correct message that indicate in log that we have exceeded the rate limit or the default message : "Cannot perform virus scan"   (VirusTotalVirusScanner.java:92)

## 3) Actual behavior

JSON Exception 

## 4) Steps to reproduce
create a fileupload page with performVirusScan=true , create a  free account on Virustotal try to scan more than 4 files in one minute

### PR DESCRIPTION
**Pull request by melloware**

### CODE REVIEW

**Comment by Rapster** [comment]
From the [API](https://developers.virustotal.com/reference), a body message is returned only if 200 http status code is returned (the error occured then because it's empty). Being said, we should cover every status code, so I like what you did, though 
1. you should use same error messages
2. extract the switch/case into a method

File: src/main/java/org/primefaces/virusscan/impl/VirusTotalVirusScanner.java [hunk 1]
```diff
@@ -70,7 +72,25 @@ public void performVirusScan(InputStream inputStream) throws VirusException {
...
+                case 204:
+                    throw new FacesException("Request rate limit exceeded. You are making more requests than allowed.");
+                case 400:
+                    throw new FacesException("Bad request. Your request was somehow incorrect.");
+                case 403:
+                    throw new FacesException("Forbidden. You don't have enough privileges to make the request.");
+                default:
+                    throw new FacesException("Unexpected HTTP code " + code + " calling Virus Total web service.");
+            }
+
```
Comment (Rapster) [inline src/main/java/org/primefaces/virusscan/impl/VirusTotalVirusScanner.java]: To be extracted into a new method, `openVirusTotalConnection`?

**Comment by Rapster** [comment]
by the way there is v3.0 available

**Comment by melloware** [comment]
Comments implemented.  I think w should wait on API 3.0 until PF 9.0.  I think @tandraschko is trying to minimize changes before 8.0 release.

**Comment by Rapster** [comment]
Yup I know i'm just saying there is a V3.0 (the API looks quite different, so definitely not in 8.0)

**Review by Rapster — APPROVED** [review]

**Pull request merged**